src/Farmer/Builders/Builders.ContainerGroups.fs

```diff
@@ -139,6 +144,18 @@ type ContainerGroupBuilder() =
         let newVolumes = volumes |> Map.ofSeq
         let updatedVolumes = state.Volumes |> Map.fold (fun current key vol -> Map.add key vol current) newVolumes
         { state with Volumes = updatedVolumes }
+    /// Sets the managed identity on this container group.
+    [<CustomOperation "identity">]
+    member _.Identity(state:ContainerGroupConfig, identity:ContainerGroupIdentity) =
+        { state with Identity = Some identity }
+    /// Enables a system assigned managed identity to be set for this container group.
+    [<CustomOperation "system_assigned_identity">]
+    member _.SystemAssignedIdentity(state:ContainerGroupConfig) =
+        { state with Identity = Some SystemAssigned }
+    /// Sets the user assigned managed identity on this container group to a user assigned identity in the same resource group.
+    [<CustomOperation "user_assigned_identity">]
+    member _.UserAssignedIdentity(state:ContainerGroupConfig, userIdentity:ResourceName) =
+        { state with Identity = Some (UserAssigned [ ManagedIdentity.UserAssignedIdentity(userIdentity.Value, None) ]) }
```

Comment:
Why do we store the name of the user assigned identity as a string, only to wrap it back into a Resource Name later, when it's actually coming in here as a resource name?